test/Test/Toml/BiCode/Property.hs

```diff
@@ -204,3 +248,40 @@ genRec = do
     btrBoolSet <- fromList <$> Gen.list (Range.constant 0 5) genBool
     btrNewtypeList <- Gen.list (Range.constant 0 5) genSum
     pure BigTypeRecord{..}
+
+----------------------------------------------------------------------------
+-- Orphan Instances
+----------------------------------------------------------------------------
+
+instance HasCodec All where
+    hasCodec = Toml.diwrap . hasCodec @Bool
+
+instance HasCodec Any where
+    hasCodec = Toml.diwrap . hasCodec @Bool
+
+instance HasCodec a => HasCodec (Sum a) where
+    hasCodec = Toml.diwrap . hasCodec @a
+
+instance HasCodec a => HasCodec (Product a) where
+    hasCodec = Toml.diwrap . hasCodec @a
+
+instance HasCodec a => HasCodec (First a) where
+    hasCodec = Toml.diwrap . hasCodec @(Maybe a)
+
+instance HasCodec a => HasCodec (Last a) where
+    hasCodec = Toml.diwrap . hasCodec @(Maybe a)
+
+instance HasItemCodec Char where
+    hasItemCodec = Left _Char
```

Comment:
This instance is weird and the fact that it's needed is caused by some unfortunate circumstances in the ecosystem. It took me some time to figure out why it's required and the reason is the following:

1. `BigType` has a field of type `String`
2. `String` is `[Char]`
3. In Tomland we have following instance:

https://github.com/kowainik/tomland/blob/d906bd9e6d63b8063955fa7382da6fe7f10f6461/src/Toml/Generic.hs#L354-L357

`String` always causes some problems, and in real life it shouldn't be a problem actually, since you always should use either `Text` or `ByteString`, but this is a bit problematic for test. I propose the following action at the moment:

1. Move `_Char` and all Char-related codes under `where` in this `instance`.
2. Mark it with `TODO` to figure out how to get rid of it later.